File: scratch/extract_logic.py
import json
import os

file_path = r'c:\Users\user\Downloads\Backened\Gniderton_Sales_Application.json'
with open(file_path, 'r', encoding='utf-8') as f:
    data = json.load(f)

# Retool JSON structure is often deeply nested. 
# It seems to be in page.data.appState which is a stringified transit-json or similar.
# But wait, looking at the previous output, it looks like standard JSON with some transit-like strings inside.

def extract_logic(obj, results):
    if isinstance(obj, dict):
        if 'id' in obj and ('query' in obj or 'funcBody' in obj):
            name = obj.get('id')
            logic = obj.get('query') or obj.get('funcBody')
            results[name] = logic
        for key, value in obj.items():
            extract_logic(value, results)
    elif isinstance(obj, list):
        for item in obj:
            extract_logic(item, results)

results = {}
extract_logic(data, results)

output_path = r'c:\Users\user\Downloads\Backened\extracted_retool_logic.txt'
with open(output_path, 'w', encoding='utf-8') as f:
    for name, logic in results.items():
        f.write(f"--- {name} ---\n{logic}\n\n")

print(f"Extracted {len(results)} logic blocks.")

File: scratch/extract_logic.js
const fs = require('fs');

const filePath = 'c:\\Users\\user\\Downloads\\Backened\\Gniderton_Sales_Application.json';
const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));

const results = {};

function extractLogic(obj) {
    if (typeof obj === 'object' && obj !== null) {
        if (obj.id && (obj.query || obj.funcBody)) {
            const name = obj.id;
            const logic = obj.query || obj.funcBody;
            results[name] = logic;
        }
        for (const key in obj) {
            extractLogic(obj[key]);
        }
    } else if (Array.isArray(obj)) {
        obj.forEach(extractLogic);
    }
}

extractLogic(data);

const outputPath = 'c:\\Users\\user\\Downloads\\Backened\\extracted_retool_logic.txt';
let outputText = '';
for (const name in results) {
    outputText += `--- ${name} ---\n${results[name]}\n\n`;
}

fs.writeFileSync(outputPath, outputText);
console.log(`Extracted ${Object.keys(results).length} logic blocks.`);
